### DISCUSSION TITLE
How add new label to textcat model

### DISCUSSION THREAD

**Discussion by info2000** [explosion/spaCy#13156] [Help: Coding & Implementations] [answered]
Hi, I'm trying to add new labels to textcat model with this config
`{'factory': 'textcat', 'model': {'@architectures': 'spacy.TextCatEnsemble.v2', 'nO': None, 'linear_model': {'@architectures': 'spacy.TextCatBOW.v2', 'exclusive_classes': True, 'ngram_size': 1, 'no_output_layer': False, 'nO': None}, 'tok2vec': {'@architectures': 'spacy.Tok2VecListener.v1', 'width': '${components.tok2vec.model.encode.width}', 'upstream': '*'}}, 'scorer': {'@scorers': 'spacy.textcat_scorer.v1'}, 'threshold': 0.5}`

but I'm getting this error:
_ValueError: [E922] Component 'textcat' has been initialized with an output dimension of 306 - cannot add any more labels_

I readed that with  TextCatBOW.v2 model is resizable but I don't know what I'm doing wrong

thanks

**Comment by svlandeg** [comment, marked as answer]
Hi! `spacy.TextCatBOW.v2` and `spacy.TextCatCNN.v2` are indeed resizable, but `spacy.TextCatEnsemble.v2` isn't and that's the main architecture of your `textcat` component. If you need a resizable textcat component, you should swap out the ensemble architecture to use either BOW or the CNN.
